jardin/cache_stores/s3.py

```diff
@@ -69,9 +77,8 @@ def __len__(self):
         return len(self.keys())
 
     def keys(self):
-        s3_client = boto3.client('s3')
         try:
-            s3_objects = s3_client.list_objects_v2(Bucket=self.bucket_name, Prefix=self.path)['Contents']
+            s3_objects = self.s3.meta.client.list_objects_v2(Bucket=self.bucket_name, Prefix=self.path)['Contents']
```

Comment:
You can derive the low-level S3 client from the high-level resource. See boto docs for [ServiceResource.meta](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/resources.html#boto3.resources.base.ServiceResource.meta).